src/avt/Database/Database/avtGenericDatabase.C

```diff
@@ -253,33 +253,47 @@ avtGenericDatabase::SetCycleTimeInDatabaseMetaData(avtDatabaseMetaData *md, int
 //      during a -dump run.
 //
 //  Created: Mark C. Miller, Wed Dec 12 04:58:26 PST 2018
+//
+//  Modifications:
+//    Kathleen Biagas, Wed May 20 14:27:24 PDT 2020
+//    Modified to dump out every domain.
+//
+//    Kathleen Biagas, Fri May 22 15:21:19 PDT 2020
+//    Handle parallel.
+//
 // ****************************************************************************
+
 static void
 DebugDumpDatasetCollection(avtDatasetCollection &dsc, int ndoms,
     string phaseName)
-
 {
     static int call_count = 0;
-    std::ostringstream oss;
 
     if (!avtDebugDumpOptions::DumpEnabled())
         return;
 
     string dumpDir = avtDebugDumpOptions::GetDumpDirectory();
     if (dumpDir == "")
         dumpDir = ".";
-    oss << dumpDir << "/gdb." << std::setfill('0') << std::setw(4) << call_count << "." << phaseName << ".vtk";
-    string dumpFile = oss.str();
-    vtkDataSetWriter *dsw = vtkDataSetWriter::New();
+    vtkNew<vtkDataSetWriter> dsw;
     dsw->SetFileTypeToASCII();
-    dsw->SetFileName(dumpFile.c_str());
     for (int i = 0 ; i < ndoms; i++)
     {
-        vtkDataSet *ds = dsc.GetDataset(i, 0);
-        dsw->SetInputData(i, ds);
+        std::ostringstream oss;
+#ifdef PARALLEL
+        oss << dumpDir << "/gdb." << std::setfill('0') << std::setw(4)
+            << call_count << "." << phaseName << ".dom"
+            << std::setfill('0') << std::setw(4) << i
+            << ".proc" << std::setfill('0') << std::setw(4) << PAR_Rank() << ".vtk";
+#else
+        oss << dumpDir << "/gdb." << std::setfill('0') << std::setw(4)
+                       << call_count << "." << phaseName << ".dom"
```

Comment:
The tabbing here is a little off. The code right above (285-287) is aligned with with the first `<<` and the code here is align with the second `<<`. Not sure which is the better choice, but I think they should be the same.